src/dispatch/plugins/dispatch_slack/service.py

```diff
@@ -100,24 +131,19 @@ def make_call(client: WebClient, endpoint: str, **kwargs) -> SlackResponse:
 
         error = exception.response["error"]
         if error == SlackAPIErrorCode.FATAL_ERROR:
-            # NOTE we've experienced a wide range of issues when Slack's performance is degraded
-            log.error(message)
-            time.sleep(300)
-            raise TryAgain from None
+            log.warn(message)
+            raise SlackRetryException(5) from None
+
         elif exception.response.headers.get("Retry-After"):
             wait = int(exception.response.headers["Retry-After"])
-            log.info(f"SlackError: Rate limit hit. Waiting {wait} seconds.")
-            time.sleep(wait)
-            raise TryAgain from None
+            log.warn(f"SlackError: Rate limit hit. Waiting {wait} seconds.")
+            raise SlackRetryException(wait) from None
+
+        # fatal error, don't retry
         raise exception
-    except Timeout as exception:
+    except (TimeoutError, Timeout) as exception:
         log.warn(f"Timeout error {exception} for slack. Endpoint: {endpoint}. Kwargs: {kwargs}")
-        time.sleep(300)
-        raise TryAgain from None
-    except TimeoutError as exception:
-        log.warn(f"TimeoutError {exception} for slack. Endpoint: {endpoint}. Kwargs: {kwargs}")
-        time.sleep(300)
-        raise TryAgain from None
+        raise SlackRetryException(5) from None
```

Comment:
```suggestion
        raise SlackRetryException(300) from None
```

Reply:
I think 300s (5min) is likely much to high for this type of operation. I'd be worried that the slack API going down would soak up threads on the server and could cause issues as these operations are blocking.